File: AndroidXPatternsApp/jmvvm-app/src/main/java/com/woowrale/jmvvm/ui/search/remote/RemoteSearchView.java
package com.woowrale.jmvvm.ui.search.remote;

import com.woowrale.jmvvm.data.model.Contact;
import com.woowrale.jmvvm.ui.adapters.ContactsAdapter;
import com.woowrale.jmvvm.ui.base.BasePresenter;

import java.util.List;

import io.reactivex.subjects.PublishSubject;

public interface RemoteSearchView extends BasePresenter.View {

    ContactsAdapter getmAdapter();

    List<Contact> getContactsList();

    PublishSubject<String> getPublishSubject();

}

File: KInstantSearch/app/src/main/java/com/woowrale/mvp/ui/search/remote/RemoteSearchView.kt
package com.woowrale.mvp.ui.search.remote

import com.woowrale.mvp.data.model.Contact
import com.woowrale.mvp.ui.base.BasePresenter
import com.woowrale.mvp.ui.adapters.ContactsAdapter
import io.reactivex.subjects.PublishSubject

interface RemoteSearchView : BasePresenter.View {

    val publishSubject: PublishSubject<String>

    fun  setContactsList(contacts: List<Contact>)

    fun getmAdapter(): ContactsAdapter

}
